src/swoop/api/io.py

```diff
@@ -0,0 +1,107 @@
+import io
+import logging
+
+from minio import Minio
+from minio.error import S3Error
+
+logger = logging.getLogger("uvicorn")
+logger.setLevel(logging.INFO)
+
+
+class IOClient:
+    client = None
+    bucket_name = None
+
+    def __init__(
+        self, s3_endpoint: str, access_key: str, secret_key: str, bucket_name=None
+    ):
+        """Initialize IO Client."""
+        self.client = Minio(
+            s3_endpoint.split("//").pop() if "http" in s3_endpoint else s3_endpoint,
+            access_key=access_key,
+            secret_key=secret_key,
+            secure="http://" not in s3_endpoint,
+        )
+        self.bucket_name = bucket_name
```

Comment:
Per our discussion, it would make sense to make `bucket_name` required and validate that it is not an empty string here. As well, you can check for the bucket and create it if it doesn't exist, so all other methods have some expectation that the bucket exists and don't have to check themselves.

```suggestion
        self, s3_endpoint: str, access_key: str, secret_key: str, bucket_name: str
    ):
        """Initialize IO Client."""
        if not bucket_name:
            raise ValueError("Bucket name cannot be empty")
        
        self.client = Minio(
            s3_endpoint.split("//").pop() if "http" in s3_endpoint else s3_endpoint,
            access_key=access_key,
            secret_key=secret_key,
            secure="http://" not in s3_endpoint,
        )
        self.bucket_name = bucket_name
        self.create_bucket()
```